server/routes/sessionList/pagedPrisonerSummaryPrisonerSession.ts

```diff
@@ -0,0 +1,259 @@
+      this.setCurrentPageNumber(1)
+    }
+
+    return this
+  }
+
+  /**
+   * Function that returns a [PrisonerSummaryPrisonerSession] filter function that operates on the specified filter field
+   * and filter value.
+   */
+  private prisonerSearchSummaryFilter =
+    (filterBy: FilterBy, value: string) =>
+    (prisonerSearchSummary: PrisonerSummaryPrisonerSession): boolean => {
+      switch (filterBy) {
+        case FilterBy.NAME: {
+          const searchTerms = [
+            prisonerSearchSummary.prisonNumber.toUpperCase(), // eg: A1234BB
+            prisonerSearchSummary.firstName.toUpperCase(), // eg: JIMMY
+            prisonerSearchSummary.lastName.toUpperCase(), // eg: LIGHTFINGERS
+            `${prisonerSearchSummary.firstName}, ${prisonerSearchSummary.lastName}`.toUpperCase(), // eg: JIMMY, LIGHTFINGERS
+            `${prisonerSearchSummary.firstName} ${prisonerSearchSummary.lastName}`.toUpperCase(), // eg: JIMMY LIGHTFINGERS
+            `${prisonerSearchSummary.lastName}, ${prisonerSearchSummary.firstName}`.toUpperCase(), // eg: LIGHTFINGERS, JIMMY
+            `${prisonerSearchSummary.lastName} ${prisonerSearchSummary.firstName}`.toUpperCase(), // eg: LIGHTFINGERS JIMMY
+            `${prisonerSearchSummary.firstName.charAt(0)} ${prisonerSearchSummary.lastName}`.toUpperCase(), // eg: J LIGHTFINGERS
+          ]
+          return searchTerms.some(searchTerm => searchTerm.includes(value.trim().toUpperCase()))
+        }
+        case FilterBy.SESSION_TYPE:
+          return prisonerSearchSummary.sessionType === value
+        default:
+          return true
+      }
+    }
+
+  /**
+   * Sorts the internal array of [PrisonerSummaryPrisonerSession] records based on the specified sort field and sort order.
+   * Other than changing the order of the internal array this does not mutate the array or individual records.
+   *
+   * This sorts the entire array, not just the current page. Because this is simply a sorting / re-ordering operation
+   * it does not remove any records, therefore the current page can be maintained (though the current page may
+   * contain different records than before the sort operation was performed)
+   */
+  sort = (sortBy: SortBy, sortOrder: SortOrder): PagedPrisonerSummaryPrisonerSession => {
+    this.prisonerSummaryPrisonerSessions = this.prisonerSummaryPrisonerSessions.sort(
+      this.prisonerSearchSummariesComparator(sortBy, sortOrder),
+    )
+    return this
+  }
+
+  /**
+   * Function that returns a [PrisonerSummaryPrisonerSession] comparator function that operates on the specified sort field
+   * and sort order.
+   */
+  private prisonerSearchSummariesComparator =
+    (sortBy: SortBy, sortOrder: SortOrder) =>
+    (left: PrisonerSummaryPrisonerSession, right: PrisonerSummaryPrisonerSession): number => {
+      switch (sortBy) {
+        case SortBy.NAME: {
+          if (sortableName(left) === sortableName(right)) {
```

Comment:
Looks like each of these case statements does the same thing... Maybe (and I don't know how) a function can be extracted that does the common comparison bits?